kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/superuser-credentials-form/index.vue

```diff
@@ -0,0 +1,197 @@
+    props: {
+      submitText: {
+        type: String,
+        required: true,
+      },
+    },
+    components: {
+      onboardingForm,
+      kTextbox,
+    },
+    data() {
+      return {
+        name: this.currentName,
+        username: this.currentUsername,
+        password: this.currentPassword,
+        passwordConfirm: this.currentPassword,
+        visitedFields: [],
+      };
+    },
+    computed: {
+      nameErrorMessage() {
+        if (this.name === '') {
+          return this.$tr('nameFieldEmptyErrorMessage');
+        }
+      },
+      usernameErrorMessage() {
+        if (this.username === '') {
+          return this.$tr('usernameFieldEmptyErrorMessage');
+        }
+        if (!/^\w+$/g.test(this.username)) {
+          return this.$tr('usernameCharacterErrorMessage');
+        }
+        return '';
+      },
+      passwordErrorMessage() {
+        if (this.password === '') {
+          return this.$tr('passwordFieldEmptyErrorMessage');
+        }
+        return '';
+      },
+      passwordConfirmErrorMessage() {
+        if (this.passwordConfirm === '') {
+          return this.$tr('passwordFieldEmptyErrorMessage');
+        }
+        if (this.passwordConfirm !== this.password) {
+          return this.$tr('passwordsMismatchErrorMessage');
+        }
+        return '';
+      },
+      nameIsInvalid() {
+        return this.visitedFields.includes('name') && !!this.nameErrorMessage;
+      },
+      usernameIsInvalid() {
+        return this.visitedFields.includes('username') && !!this.usernameErrorMessage;
+      },
+      passwordIsInvalid() {
+        return this.visitedFields.includes('password') && !!this.passwordErrorMessage;
+      },
+      passwordConfirmIsInvalid() {
+        return this.visitedFields.includes('passwordConfirm') && !!this.passwordConfirmErrorMessage;
+      },
+      formIsValid() {
+        return (
+          !this.usernameIsInvalid &&
+          !this.passwordIsInvalid &&
+          !this.passwordConfirmIsInvalid &&
+          !this.facilityIsInvalid
+        );
+      },
+    },
+    methods: {
+      validate(fieldName) {
+        this.visitedFields.push(fieldName);
+      },
+      setSuperuserCredentials() {
+        const focusOnInvalidField = () => {
+          if (this.nameIsInvalid) {
+            this.$refs.name.focus();
+          } else if (this.usernameIsInvalid) {
+            this.$refs.username.focus();
+          } else if (this.passwordIsInvalid) {
+            this.$refs.password.focus();
+          } else if (this.passwordConfirmIsInvalid) {
+            this.$refs.passwordConfirm.focus();
+          }
+        };
+
+        this.validate('name');
+        this.validate('username');
+        this.validate('password');
+        this.validate('passwordConfirm');
+
+        if (this.formIsValid) {
+          this.submitSuperuserCredentials(this.name, this.username, this.password);
+          this.$emit('submit');
+        } else {
+          focusOnInvalidField();
```

Comment:
This method is only called if the form is invalid, so might as well put it in here.

Reply:
I did it this way mainly to give the set of directions a name. You think it's not worth it, in this case?

Reply:
Well in that case, I would find it a bit more clean if you added this method to the methods object.  :smile:

Reply:
I disagree :smirk: I'll just remove the function.

Reply:
:smirk: